File: cpp/src/VendingMachinePrinter.cpp

#include <sstream>
#include "VendingMachinePrinter.h"

VendingMachinePrinter::VendingMachinePrinter(VendingMachine* machine):
machine(machine), columns(60) {}



std::string VendingMachinePrinter::print() {
    std::string printout = "VendingMachine";
    // TODO: finish this

    return printout;
}

VendingMachinePrinter::~VendingMachinePrinter() {
    machine = nullptr;
}

// convenience function to lay out a key va

File: python/vending_machine_printer.py
from vending_machine import VendingMachine


class VendingMachinePrinter:
    def __init__(self, machine : VendingMachine):
        self.columns = 60
        self.machine = machine

    def print(self):
        text = "VendingMachine\n"
        # TODO: finish this
        return text

    def format_line_with_whitespace(self, name, value):
        """
        Convenience function that lays out a name and a value at either ends of a fixed-width line.
        eg if you call it with name="Foo" value="Bar" it will return

        Foo                                       Bar

        """
        va